src/runners/global-search-runner.js

```diff
@@ -4,12 +4,28 @@ import GlobalSearch from '../../components/global_search/global-search.react';
 
 // import component
 
-ReactDOM.render(
-	<GlobalSearch desktop={true}/>,
-	document.getElementById('global-search-desktop-component'),
-);
-
-ReactDOM.render(
-	<GlobalSearch desktop={false}/>,
-	document.getElementById('global-search-mobile-component'),
-);
\ No newline at end of file
+
+function renderIfExists(id, desktop, defaultFormDataIndex) {
+
+	if(document.getElementById(id)) {
+		ReactDOM.render(
+			<GlobalSearch desktop={desktop} defaultFormDataIndex={defaultFormDataIndex}
+			/>,
+			document.getElementById(id)
+		)
+	}
+
+
+}
+
+
+// Non-discovery runner
+
+renderIfExists("global-search-desktop-component", true, 0);
+renderIfExists("global-search-mobile-component", false, 0);
+
+
+// Discovery runner
+
+renderIfExists("global-search-desktop-component-discovery", true, 1);
+renderIfExists("global-search-mobile-component-discovery", false, 1);
```

Comment:
I wonder if we should add a newline here (just for consistency and to get rid of the GitHub warning icon)